packages/terra-data-grid/src/DataGrid.jsx

```diff
@@ -495,7 +502,27 @@ const DataGrid = forwardRef((props, ref) => {
     if (!event.currentTarget.contains(event.relatedTarget)) {
       // Not triggered when swapping focus between children
       if (handleFocus.current) {
-        setFocusedRowCol(focusedRow, focusedCol, true);
+        let newRowIndex = focusedRow;
+        let newColumnIndex = focusedCol;
+        const headerOffset = hasVisibleColumnHeaders ? 1 : 0;
```

Comment:
I would not expect us to need a headerOffset.  Even though the headers are not visible, they are still part of the table.  Do you have a unit test for this scenario?